src/IIIFPresentation/API/Features/Manifest/CanvasPaintingResolver.cs

```diff
@@ -160,4 +190,61 @@ public class CanvasPaintingResolver(
             return null;
         }
     }
-}
\ No newline at end of file
+    
+    private async Task<(PresUpdateResult? updateResult, List<CanvasPainting>? canvasPaintings)> CreateCanvasPaintingsFromAssets(
+        int customerId, PresentationManifest presentationManifest, int manifestSpace, CancellationToken cancellationToken)
+    {
+        var paintedResourceCount = presentationManifest.PaintedResources!.Count;
+
+        var canvasPaintingIds =
+            await GenerateUniqueCanvasPaintingIds(paintedResourceCount, customerId, cancellationToken);
+        if (canvasPaintingIds == null)
+            return (ErrorHelper.CannotGenerateUniqueId<PresentationManifest>(), null);
+        
+        var canvasPaintings = new List<CanvasPainting>();
+        int count = 0;
+        foreach (var paintedResource in presentationManifest.PaintedResources)
+        {
+            if (paintedResource.Asset == null) continue; // do we need to log this??
+    
+            if (!paintedResource.Asset.TryGetValue("space", out var space))
+            {
+                paintedResource.Asset.Add("space", manifestSpace);
+                space = manifestSpace;
+            }
+
+            if (!paintedResource.Asset.TryGetValue("id", out var id))
+            {
+                return (ErrorHelper.CouldNotRetrieveAssetId<PresentationManifest>(), null);
+            }
+
+            var cp = new CanvasPainting
+            {
+                Id = canvasPaintingIds[count],
+                Label = GetLabel(paintedResource.CanvasPainting),
+                Created = DateTime.UtcNow,
+                CustomerId = customerId,
+                CanvasOrder = count,
+                AssetId = $"{customerId}/{space}/{id}",
+                ChoiceOrder = -1
+            };
+
+            count++;
+            canvasPaintings.Add(cp);
+        }
+        return (null, canvasPaintings);
+    }
+
+    private LanguageMap GetLabel(Models.API.Manifest.CanvasPainting canvasPainting)
+    {
+        if (canvasPainting?.Label == null)
+        {
+            return new LanguageMap()
+            {
+                {"en", new List<string>(){"Created by the DLCS"}}
```

Comment:
I've modified this to not set a default label as that might be better to look at on #204, but will leave this open for comments as this might need some discussion

Reply:
Labels were mandatory in P2 but SHOULD in P3. And the "SHOULD automatically generate" is the client's task (e.g., UV, Mirador). So we should **not** auto-generate anything the user hasn't sent.